statsmodels/discrete/discrete_model.py

```diff
@@ -414,14 +416,16 @@ def _derivative_exog(self, params, exog=None, dummy_idx=None,
 
 class BinaryModel(DiscreteModel):
 
-    def __init__(self, endog, exog, **kwargs):
-        super(BinaryModel, self).__init__(endog, exog, **kwargs)
+    def __init__(self, endog, exog, offset=None, **kwargs):
+        super(BinaryModel, self).__init__(endog, exog, offset=offset, **kwargs)
         if (not issubclass(self.__class__, MultinomialModel) and
                 not np.all((self.endog >= 0) & (self.endog <= 1))):
             raise ValueError("endog must be in the unit interval.")
+        if offset is None:
+            delattr(self, 'offset')
```

Comment:
I know this is done elsewhere so it's not the absolute worst, but is there a compelling reason not to just leave `self.offset = None`?  The only difference AFAICT would be that below instead of `hasattr(self, 'offset')` the check would be `self.offset is not None`.